recaf-ui/src/main/java/me/coley/recaf/ui/control/NavigationBar.java

```diff
@@ -0,0 +1,173 @@
+import me.coley.recaf.ui.CommonUX;
+import me.coley.recaf.ui.control.menu.ActionMenuItem;
+import me.coley.recaf.ui.util.Icons;
+import me.coley.recaf.util.logging.Logging;
+import org.slf4j.Logger;
+
+/**
+ * Navigation bar implementation. Helpful for easily navigating around a class.
+ * @author yapht
+ */
+public class NavigationBar extends HBox {
+    private static final Logger logger = Logging.get(NavigationBar.class);
+
+    private static class NavigationSeparator extends Canvas {
+        public NavigationSeparator() {
+            setWidth(5);
+            setHeight(15);
+
+            double pointY = (getHeight() / 2);
+            double pointX = getWidth() - 1;
+
+            // Draw a grey chevron
+            GraphicsContext gc = getGraphicsContext2D();
+            gc.setStroke(Color.rgb(90, 90, 90));
+            gc.strokeLine(0, 0, pointX, pointY);
+            gc.strokeLine(pointX, pointY, 0, getHeight());
+        }
+    }
+
+    private static class MemberNavigationNode extends Label {
+        private static class ScrollableContextMenu extends ContextMenu {
+            public ScrollableContextMenu() {
+                addEventHandler(Menu.ON_SHOWING, e -> {
+                    Node content = getSkin().getNode();
+                    if (content instanceof Region) {
+                        ((Region) content).setMaxHeight(getMaxHeight());
+                    }
+                });
+            }
+        }
+
+        private final CommonClassInfo classInfo;
+
+        public MemberNavigationNode(String text, CommonClassInfo classInfo) {
+            super(text);
+            this.classInfo = classInfo;
+
+            setOnMouseClicked(event -> onMouseClicked());
+            setGraphic(Icons.getClassIcon(classInfo));
+        }
+
+        public MemberNavigationNode(String text, CommonClassInfo classInfo, Node icon) {
+            super(text);
+            this.classInfo = classInfo;
+
+            setOnMouseClicked(event -> onMouseClicked());
+            setGraphic(icon);
+        }
+
+        private void onMouseClicked() {
+            ScrollableContextMenu menu = new ScrollableContextMenu();
+            menu.setMaxHeight(500);
+
+            // Stupid workaround for the menu not showing at the anchor
+            menu.getItems().add(new MenuItem("hack"));
+            menu.show(this, Side.BOTTOM, 0, 5);
+            menu.getItems().clear();
+
+            for(MethodInfo method : classInfo.getMethods())
+                menu.getItems().add(new ActionMenuItem(method.getName(), Icons.getMethodIcon(method), () -> CommonUX.openMember(classInfo, method)));
+
+            for(FieldInfo field : classInfo.getFields())
+                menu.getItems().add(new ActionMenuItem(field.getName(), Icons.getFieldIcon(field), () -> CommonUX.openMember(classInfo, field)));
+        }
+    }
+
+    private NavigationBar()  {
+        setStyle(
+                "-fx-spacing: 8px;" +
+                "-fx-padding: 4 0 4 8;");
```

Comment:
Would like to extract this to a `CSS`. In 3X most control groups have their own file, so this could be either put in the bottom of `menu.css` or make a new one for `navbar.css` and add it to the main CSS files list.